tools/SendBlobs/FundsDistributor.cs

```diff
@@ -0,0 +1,192 @@
+
+            string? result = await nodeManager.Post<string>("eth_sendRawTransaction", "0x" + txRlp);
+            if (result != null)
+                txHash.Add(result);
+            nonce++;
+        }
+        if (!string.IsNullOrWhiteSpace(keyFilePath))
+            File.WriteAllLines(keyFilePath, privateKeys.Select(k => k.ToString()), Encoding.ASCII);
+
+        return txHash;
+    }
+    /// <summary>
+    /// Send all available funds from a list of private keys contained in <paramref name="keyFilePath"/> to <paramref name="beneficiary"/>.
+    /// </summary>
+    /// <param name="nodeManager"></param>
+    /// <param name="chainId"></param>
+    /// <param name="beneficiary"></param>
+    /// <param name="keyFilePath"></param>
+    /// <param name="logManager"></param>
+    /// <returns><see cref="IEnumerable{string}"/> containing all the executed tx hashes.</returns>
+    public async static Task<IEnumerable<string>> ReclaimFunds(INodeManager nodeManager, ulong chainId, Address beneficiary, string keyFilePath, ILogManager logManager, UInt256 maxFee, UInt256 maxPriorityFee)
+    {
+        IEnumerable<Signer> privateSigners =
+            File.ReadAllLines(keyFilePath, Encoding.ASCII)
+            .Select(k => new Signer(chainId, new PrivateKey(k), logManager));
+
+        ILogger log = logManager.GetClassLogger();
+        List<string> txHashes = new List<string>();
+        foreach (var signer in privateSigners)
+        {
+            string? balanceString = await nodeManager.Post("eth_getBalance", signer.Address, "latest");
+            if (balanceString is null)
+                continue;
+            string? nonceString = await nodeManager.Post<string>("eth_getTransactionCount", signer.Address, "latest");
+            if (nonceString is null)
+                continue;
+
+            UInt256 balance = new UInt256(Bytes.FromHexString(balanceString));
+
+            ulong nonce = Convert.ToUInt64(nonceString, nonceString.StartsWith("0x") ? 16 : 10);
+
+            string? gasPriceRes = await nodeManager.Post<string>("eth_gasPrice") ?? "1";
+            UInt256 gasPrice = (UInt256)Convert.ToUInt64(gasPriceRes, gasPriceRes.StartsWith("0x") ? 16 : 10);
+
+            UInt256 maxPriorityFeePerGas = maxPriorityFee;
+            if (maxPriorityFee == 0)
+            {
+                string? maxPriorityFeePerGasRes = await nodeManager.Post<string>("eth_maxPriorityFeePerGas") ?? "1";
+                maxPriorityFeePerGas = (UInt256)Convert.ToUInt64(maxPriorityFeePerGasRes, maxPriorityFeePerGasRes.StartsWith("0x") ? 16 : 10);
+            }
+
+            UInt256 approxGasFee = (gasPrice + maxPriorityFeePerGas) * GasCostOf.Transaction;
+
+            if (balance < approxGasFee)
+            {
+                log.Info($"Not enough funds on {signer.Address} to pay for gas");
+                continue;
+            }
+
+            UInt256 toSend = balance - approxGasFee;
+
+            TxDecoder txDecoder = new();
```

Comment:
We don't need a separate decoder for each transaction, we can create it just once.

Reply:
Fixed.